src/box/replication.cc

```diff
@@ -1160,6 +1197,8 @@ replicaset_connect(const struct uri_set *uris,
 	/* Now all the appliers are connected, update the replica set. */
 	replicaset_update(appliers, count, keep_connect);
 	appliers_guard.is_active = false;
+	replication_fiber = NULL;
+	fiber_cond_signal(&replication_connect_cond);
```

Comment:
Probably, better add another guard that wouldn't be disarmed on success to signal this condition variable.

Reply:
Done.

```diff
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -1089,6 +1089,10 @@ replicaset_connect(const struct uri_set *uris,
        }
        assert(replication_connect_fiber == NULL);
        replication_connect_fiber = fiber();
+       auto connect_fiber_guard = make_scoped_guard([&]{
+               replication_connect_fiber = NULL;
+               fiber_cond_signal(&replication_connect_cond);
+       });
        int count = 0;
        struct applier *appliers[VCLOCK_MAX] = {};
        auto appliers_guard = make_scoped_guard([&]{
@@ -1097,8 +1101,6 @@ replicaset_connect(const struct uri_set *uris,
                        applier_stop(applier);
                        applier_delete(applier);
                }
-               replication_connect_fiber = NULL;
-               fiber_cond_signal(&replication_connect_cond);
        });
        for (; count < uris->uri_count; count++)
                appliers[count] = applier_new(&uris->uris[count]);
@@ -1204,8 +1206,6 @@ replicaset_connect(const struct uri_set *uris,
        /* Now all the appliers are connected, update the replica set. */
        replicaset_update(appliers, count, keep_connect);
        appliers_guard.is_active = false;
-       replication_connect_fiber = NULL;
-       fiber_cond_signal(&replication_connect_cond);
        uri_set_destroy(&replication_uris);
        uri_set_copy(&replication_uris, uris);
 }
```